File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
import java.util.*;
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int numer = numer1 * denom2 + numer2 * denom1;
        int denom = denom1 * denom2;

        // 최대공약수 구하기
        int gcd = gcd(numer, denom);

        numer /= gcd;
        denom /= gcd;

        return new int[]{numer, denom};
    }

    // 최대공약수 계산
    public int gcd(int a, int b) { // 10/8
        while (b != 0) {
            int r = a % b; //10%8 == 2, 8%2==0
            a = b;
            b = r;
        }
        return a;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
    
    
    let numer = numer1*denom2 + numer2*denom1;
    let denom = denom1*denom2;
    
    let a=numer,b=denom;
    while(b!=0) {
        let r = a%b;
        a = b;
        b = r;
    }
    numer/=a;
    denom/=a;
    
    const answer = [numer,denom];
    
    return answer;
}